druid-shell/src/platform/windows/dialog.rs

```diff
@@ -119,7 +121,27 @@ pub(crate) unsafe fn get_file_dialog_path(
 
     if let Some(spec) = &raw_spec {
         as_result(file_dialog.SetFileTypes(spec.len() as u32, spec.as_ptr()))?;
-        as_result(file_dialog.SetFileTypeIndex(1))?;
+
+        let index = match &options.default_type {
+            Some(default_type) => options
+                .allowed_types
+                .as_ref()
+                .unwrap()
+                .iter()
+                .position(|allowed_type| allowed_type == default_type)
+                .or(Some(0))
+                .unwrap(),
+            None => 0,
```

Comment:
I think this should be `1`, and the position above should be incremented by 1? I might be wrong, i'm just basing this off the [`IFileDialog::SetFileTypeIndex` docs](https://docs.microsoft.com/en-us/previous-versions/windows/desktop/legacy/bb775978(v%3Dvs.85)#parameters).

Reply:
`SetFileTypeIndex` wants 1-based indeed and it gets that +1 when it is called. The `index` variable otherwise refers to the Rust vector index of `options.allowed_types`. I think it makes sense to keep the index 0-based as that is what the druid code uses and then do the +1 when doing the external `SetFileTypeIndex` call. Especially because `index` is also used where it needs to be 0-based.